spec/features/user_adds_video_to_queue_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+
+
+feature 'user interacts with the queue' do
+  
+  scenario 'user adds videos to and reorders queue' do
+
+    videos = []
+    category  = Fabricate(:category)
+    videos[0] = Fabricate(:video, category: category)
+    videos[1] = Fabricate(:video, category: category)
+    videos[2] = Fabricate(:video, category: category)
+
+    sign_in_user
+
+    add_videos_to_queue(videos)
+    reorder_queue(videos)
+    update_queue
+    verify_queue(videos)
+
+  end
+
+  def update_queue
+    click_button "Update Instant Queue"
+  end
+
+  def add_videos_to_queue(videos)
+    videos.each do |video|
+      add_video_to_queue(video)
+      verify_hidden_button(video)
+    end
+  end
+
+  def add_video_to_queue(video)
+    expect(page.current_path).to eq "/home"
+    find("a[href='/videos/#{video.id}']").click
+    expect(page.current_path).to eq "/videos/#{video.id}"
+    click_link "+ My Queue"
+    expect(page.current_path).to eq "/queue_items"
+    expect(page).to have_content "#{video.title}"
+  end
+  
+  def verify_hidden_button(video)
+    find("a[href='/videos/#{video.id}']").click
+    expect(page.current_path).to eq "/videos/#{video.id}"
+    expect(page).not_to have_content "+ My Queue"
```

Comment:
I would rather use `expect(page).to have_no_content "some text"`. It works with ajax better, which might be around in future. In general `have_no_content` is considered a slightly better than `.not_to have_content`.